ngraph/core/CMakeLists.txt

```diff
@@ -70,6 +75,9 @@ if(NGRAPH_LIB_VERSIONING_ENABLE)
         SOVERSION ${NGRAPH_API_VERSION})
 endif()
 target_compile_definitions(ngraph PUBLIC NGRAPH_VERSION="${NGRAPH_VERSION}")
+if(NGRAPH_STATIC_LIBRARY)
+    target_compile_definitions(ngraph PUBLIC NGRAPH_STATIC_LIBRARY)
```

Comment:
Looks like we should't use `target_compile_definitions` for subfolders after the [line](https://github.com/openvinotoolkit/openvino/pull/2856/files#diff-3d2b555bea347adb474b05f13a23deef8ff07c2fb21bc167a5f90a09ea16b796R410), or in other case we should't add global definitions.